### PR TITLE
ERC721 metadata subgraph

### PR DESCRIPTION
**Pull request by corerouter**

### CODE REVIEW

File: subgraphs/erc721-metadata/schema.graphql [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+  id: ID!
+
+  " Address of the collection  "
+  collection: String!
+
+  " ID of the token from the collection  "
+  tokenId: BigInt!
+
+  " Token metadata this attribute belongs to "
+  token: Token
```
Comment (0xbe1) [inline subgraphs/erc721-metadata/schema.graphql:59]: should this be required?

File: subgraphs/erc721-metadata/schema.graphql [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+
+  " Indicates how the trait should be displayed. "
+  displayType: String
+}
+
+#####################################
+####### ERC721 Collections ######
+#####################################
+
+type Collection @entity {
```
Comment (0xbe1) [inline subgraphs/erc721-metadata/schema.graphql:75]: maybe name it ERC721Collection

**Review by 0xbe1 — DISMISSED** [review]
lgtm

File: subgraphs/erc721-metadata/schema.graphql (hunk 1 above)
Comment (corerouter) [reply subgraphs/erc721-metadata/schema.graphql]: Changed in latest check-in 1d147c5814076e3b053459ebda09b35c60ff90b9

File: subgraphs/erc721-metadata/schema.graphql (hunk 2 above)
Comment (corerouter) [reply subgraphs/erc721-metadata/schema.graphql]: Changed in 1d147c5814076e3b053459ebda09b35c60ff90b9 Thanks!

**Review by mrbrianhobo — DISMISSED** [review]
this also looks good to me 👍

File: subgraphs/erc721-metadata/schema.graphql [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+  description: String
+
+  " A URI pointing to a resource with mime type image, defined in opensea metadata standards "
+  imageURI: String
+
+  " URL which allows users to view the item on the colletion's site, defined in opensea metadata standards "
+  externalURI: String
+
+  " Attributes for the token metadata, defined in opensea metadata standards "
+  attributes: [Trait!] @derivedFrom(field: "token")
```
Comment (this-username-is-taken) [inline subgraphs/erc721-metadata/schema.graphql:35]: why not call this `traits` to match with the name of the entity?

**Comment by this-username-is-taken** [comment]
There is a bit more metadata on OpenSea you might want to add: https://docs.opensea.io/docs/metadata-standards#metadata-structure

File: subgraphs/erc721-metadata/schema.graphql [hunk 4]
```diff
@@ -0,0 +1,93 @@
...
+
+  " Block number of the metadata snapshot "
```
Comment (this-username-is-taken) [inline subgraphs/erc721-metadata/schema.graphql:37]: What does this mean?

File: subgraphs/erc721-metadata/schema.graphql [hunk 5]
```diff
@@ -0,0 +1,93 @@
...
+  " Collection for the token "
+  collection: ERC721Collection!
+
+  " ID of the token from the collection "
+  tokenID: BigInt!
+
+  " URI of the token metadata"
+  tokenURI: String
+
+  " Identifies the asset to which this token represents, defined in opensea metadata standards "
```
Comment (this-username-is-taken) [inline subgraphs/erc721-metadata/schema.graphql:22]: nit: change `opensea` to `OpenSea`

File: subgraphs/erc721-metadata/schema.graphql [hunk 6]
```diff
@@ -0,0 +1,93 @@
...
+
+  " Indicates how the trait should be displayed. "
+  displayType: String
+}
+
+#####################################
+####### ERC721 Collection ######
+#####################################
+
+type ERC721Collection @entity {
```
Comment (this-username-is-taken) [inline subgraphs/erc721-metadata/schema.graphql:75]: Why `ERC721Collection`? Shouldn't this be consistent with https://github.com/messari/subgraphs/blob/76ccecf4361879c73e9ec4066af9edb4e33cf47a/subgraphs/erc721/schema.graphql#L33?

File: subgraphs/erc721-metadata/schema.graphql [hunk 7]
```diff
@@ -0,0 +1,93 @@
...
+  " ERC721 collection address "
+  id: ID!
+
+  " Flag for whether ERC721 metadata is supported "
+  supportsERC721Metadata: Boolean!
+
+  " Tokens in the collection "
+  tokens: [Token!]! @derivedFrom(field: "collection")
```
Comment (this-username-is-taken) [inline subgraphs/erc721-metadata/schema.graphql:83]: we should also have a `tokenCount: BigInt!`

**Comment by this-username-is-taken** [comment]
Some minor comments, otherwise looks all good!

File: subgraphs/erc721-metadata/schema.graphql (hunk 4 above)
Comment (corerouter) [reply subgraphs/erc721-metadata/schema.graphql:37]: Modify comments at 7994d3b6826bfe9f90d4e200b29e7d05f390ca38

File: subgraphs/erc721-metadata/schema.graphql (hunk 6 above)
Comment (corerouter) [reply subgraphs/erc721-metadata/schema.graphql:75]: Modified in f8c3cf85b80323f92ec418ea8840d9563e47f3b4

File: subgraphs/erc721-metadata/schema.graphql (hunk 7 above)
Comment (corerouter) [reply subgraphs/erc721-metadata/schema.graphql:83]: Add in f8c3cf85b80323f92ec418ea8840d9563e47f3b4

**Comment by this-username-is-taken** [comment]
Seems like there are two subgraph folders. Probably should clean that up.

**Comment by corerouter** [comment]
Sorry to forget to cleanup in the repository after changing folder name locally. Corrected it in [95b99a6](https://github.com/messari/subgraphs/pull/725/commits/95b99a68010eec655294899a087e204a6736fe28)

**Comment by this-username-is-taken** [comment]
Great work!

**Pull request merged**